packages/rocketchat-katex/katex.js

```diff
@@ -0,0 +1,254 @@
+
+		// Search for a closer delimiter after the opening one
+		const closer_index = str.substr(inner.start).indexOf(closer);
+		if (closer_index < 0) {
+			return null;
+		}
+		inner.end = inner.start + closer_index;
+		outer.end = inner.end + closer.length;
+		return {
+			outer,
+			inner
+		};
+	}
+
+	// Searches for the first latex block in the given string
+	find_latex(str) {
+		let start = 0;
+		let opening_delimiter_match;
+
+		while ((opening_delimiter_match = this.find_opening_delimiter(str, start++)) != null) {
+			const match = this.get_latex_boundaries(str, opening_delimiter_match);
+			if (match && match.inner.extract(str).trim().length) {
+				match.options = opening_delimiter_match.options;
+				return match;
+			}
+		}
+		return null;
+	}
+
+	// Breaks a message to what comes before, after and to the content of a
+	// matched latex block
+	extract_latex(str, match) {
+		const before = str.substr(0, match.outer.start);
+		const after = str.substr(match.outer.end);
+		let latex = match.inner.extract(str);
+		latex = s.unescapeHTML(latex);
+		return {
+			before,
+			latex,
+			after
+		};
+	}
+
+	// Takes a latex math string and the desired display mode and renders it
+	// to HTML using the KaTeX library
+	render_latex(latex, displayMode) {
+		let rendered;
+		try {
+			rendered = katex.renderToString(latex, {
+				displayMode
+			});
+		} catch (error) {
+			const e = error;
+			const display_mode = displayMode ? 'block' : 'inline';
+			rendered = `<div class="katex-error katex-${ display_mode }-error">`;
+			rendered += `${ s.escapeHTML(e.message) }`;
+			rendered += '</div>';
+		}
+		return rendered;
+	}
+
+	// Takes a string and renders all latex blocks inside it
+	render(str, render_func) {
+		let result = '';
+		while (this.find_latex(str) != null) {
+			// Find the first latex block in the string
+			const match = this.find_latex(str);
+			const parts = this.extract_latex(str, match);
+
+			// Add to the reuslt what comes before the latex block as well as
+			// the rendered latex content
+			const rendered = render_func(parts.latex, match.options.displayMode);
+			result += parts.before + rendered;
+			// Set what comes after the latex block to be examined next
+			str = parts.after;
+		}
+		return result += str;
+	}
+
+	// Takes a rocketchat message and renders latex in its content
+	render_message(message) {
+		//Render only if enabled in admin panel
+		let render_func;
+		if (this.katex_enabled()) {
+			let msg = message;
+			if (!_.isString(message)) {
+				if (_.trim(message.html)) {
+					msg = message.html;
+				} else {
+					return message;
+				}
+			}
+			if (_.isString(message)) {
+				render_func = (latex, displayMode) => {
+					return this.render_latex(latex, displayMode);
+				};
+			} else {
+				if (message.tokens == null) {
+					message.tokens = [];
+				}
+				render_func = (latex, displayMode) => {
+					const token = `=&=${ Random.id() }=&=`;
```

Comment:
We are not using `=&=` anymore, use `=!=`. Be careful when you sync, probably you got a conflict here and ignored the change